pkg/occlient/occlient_test.go

```diff
@@ -1398,20 +1398,22 @@ func TestCreateServiceBinding(t *testing.T) {
 		name        string
 		bindingNS   string
 		bindingName string
+		labels      map[string]string
 		wantErr     bool
 	}{
 		{
 			name:        "Case: Valid request for creating a secret",
```

Comment:
I think we should have some more test case for this unit test like we should have the test case for ```invalid request ``` also. What will happen if we pass ```empty labels``` or ```different binding name``` other than ```foo```. I think we should consider these test case also. It's just a thought, WDYT?

Reply:
I looked at it as you suggested. But we're getting these values from the function that calls `CreateServiceBinding`. About the invalid request point, it's valid but `CreateServiceBinding` uses function in the OpenShift API to create the Service Binding. So adding a test for it would be futile (assuming the tests already exist in OpenShift source code.)

Reply:
IMO we should also add some invalid test case to be safe. Depending on Openshift source code tests won't be good idea i guess. :-)

Reply:
> I think we should have some more test case for this unit test like we should have the test case for `invalid request ` also. What will happen if we pass `empty labels` or `different binding name` other than `foo`. I think we should consider these test case also. It's just a thought, WDYT?

The function we are testing here just calls the openshift api server with the arguments passed to it. The cases that you have mentioned would have been already taken care of by openshift.

It's not about scope but actually who is responsible of the test. once the occlient handed over the data all the validation and functionality come under the scope of openshift. and so does testing those pieces.